### PR TITLE
Remove pickle python 2 library

### PR DESCRIPTION
**Pull request by h-mayorquin**
I encountered this today while debuggin some issue.

Python 2 is long gone:
https://devguide.python.org/versions/

### CODE REVIEW

File: neo/io/pickleio.py [hunk 1]
```diff
@@ -10,10 +10,8 @@
 Authors: Andrew Davison
 """
 
-try:
-    import cPickle as pickle  # Python 2
-except ImportError:
-    import pickle  # Python 3
+
+import pickle  # Python 3
```
Comment (JuliaSprenger) [inline neo/io/pickleio.py:14]:
```suggestion
import pickle
```

**Comment by JuliaSprenger** [comment]
Good catch! Thanks for the PR!

**Comment by JuliaSprenger** [comment]
Failing tests are not related to the changes in the pickleIO

**Comment by h-mayorquin** [comment]
I was very confused about the failing tests : P

**Pull request merged**